swap/src/bin/swap.rs

```diff
@@ -76,7 +77,14 @@ async fn main() -> Result<()> {
                 init_monero_wallet(data_dir, monero_daemon_host, env_config).await?;
             let bitcoin_wallet = Arc::new(bitcoin_wallet);
 
-            let mut swarm = swarm::bob(&seed, alice_peer_id)?;
+            let tor_socks5_port = {
+                let uac =
+                    tor::UnauthenticatedConnection::default().with_socks5_port(tor_socks5_port);
+                uac.assert_tor_running()
+                    .await
+                    .map_or(None, |_| Some(uac.tor_proxy_port()))
+            };
```

Comment:
This is duplicated between two match arms. I think this could also just move into the `swarm::bob` function given that it already "knows" about the socks5 port.